### ISSUE TITLE
Segment did not perform as expected, querynode memory growth was uneven

### ISSUE THREAD

**Issue by yesletgo** [milvus-io/milvus#23477]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.5
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):   pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2):2.2.3
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 84cores/600G
- GPU: 
- Others:
```

### Current Behavior

![1681782630073](https://user-images.githubusercontent.com/21998052/232649547-c90dbf88-b005-482e-b0e9-3446732fe7b3.jpg)
<img width="459" alt="1681782697988" src="https://user-images.githubusercontent.com/21998052/232649571-a2b58844-6522-40d4-8aae-04121d780447.png">

**After manual flush**
![1681782786265](https://user-images.githubusercontent.com/21998052/232649589-3a159d6d-e0f8-400a-ae35-cfdca5a414ae.jpg)

### Milvus Log

https://video-tag-lib-1251808348.cos.ap-beijing.myqcloud.com/milvus_log/milvus-log.tar.gz

**Comment by yanliang567** [comment]
/assign @congqixia 
it is weird that there are many segments with row_count=0
/unassign

**Comment by congqixia** [comment]
@yesletgo from the screen shots and the log, it looks like the datanode cannot watch channel in time. It's a known issue and shall be fixed in 2.2.6
You could update this configuration as a quick workaround:
```yaml
# milvus.yaml
dataCoord:
...
  channel:
    watchTimeoutInterval:  120 # default value is 30
```

**Comment by yesletgo** [comment]
> @yesletgo from the screen shots and the log, it looks like the datanode cannot watch channel in time. It's a known issue and shall be fixed in 2.2.6 You could update this configuration as a quick workaround:
> 
> ```yaml
> # milvus.yaml
> dataCoord:
> ...
>   channel:
>     watchTimeoutInterval:  120 # default value is 30
> ```
The configuration has been updated and the datacoord has been restarted，but the phenomenon is still the same.
![image](https://user-images.githubusercontent.com/21998052/232750838-189b58b3-db61-40a2-a9fb-4121c03873d7.png)

[milvus-log2.tar.gz](https://github.com/milvus-io/milvus/files/11260703/milvus-log2.tar.gz)
